network/basics/uri.py

```diff
@@ -40,6 +40,12 @@
         directory, the basename of the file on the remote server will be used.
     required: false
     default: null
+  source:
+    description:
+      - path to binary file to be uploaded to the endpoint
```

Comment:
Assuming that you mark this mutually_exclusive with body, you should mention that you can only have one or the other in the documentation as well.